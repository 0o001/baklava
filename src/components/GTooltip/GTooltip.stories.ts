import { storiesOf } from '@storybook/vue';
import { withKnobs, select, boolean, text } from '@storybook/addon-knobs';
import GTooltip from './GTooltip.vue';
import GText from '../GText/GText.vue';

storiesOf('GTooltip', module)
  .addDecorator(withKnobs)
  .addParameters({ component: GTooltip })
  .add('base', () => ({
    components: { GTooltip, GText },
    template: `
      <div style="text-align: center; margin:100px;">
        <GTooltip text="Tooltip Text">
          <template #tooltip-trigger>
            <GText>Tooltip Trigger Text</GText>
          </template>
        </GTooltip>
        </div>
      `,
  }))
  .add('variant', () => ({
    components: { GTooltip, GText },
    props: {
      variant: {
        default: select("variant", [
          "success",
          "primary",
          "secondary",
          "tertiary",
          "info",
          "warning",
          "danger",
          "light",
          "dark",
        ], "dark"),
      },
      placement: {
        default: select("placement", [
          "top",
          "left",
          "bottom",
          "right",
        ], "right"),
      },
      tooltipText: {
        default: text("tooltipText", "Trendyol Seller Center"),
      },
    },
    template: `
      <div style="text-align: center; margin:100px;">
        <GTooltip :text="tooltipText" :variant="variant" :placement="placement">
          <template #tooltip-trigger>
            <GText>Tooltip Trigger Text</GText>
          </template>
        </GTooltip>
      </div>
    `,
  }))
  .add('slot text', () => ({
    components: { GTooltip, GText },
    template: `
      <div style="text-align: center; margin:100px;">
        <GTooltip>
          <template #tooltip-trigger>
            <GText>Tooltip Trigger Text</GText>
          </template>
          <template #tooltip-text>
            Tooltip Text
          </template>
        </GTooltip>
      </div>
    `,
  }))
