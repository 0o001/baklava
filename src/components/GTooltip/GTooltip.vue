<template>
  <div class="g-tooltip">
    <slot name="tooltip-trigger" />
    <span
      class="text"
      :class="placementClass"
    >
      <i class="arrow" />
      <span>
        {{ text }}
        <slot name="tooltip-text" />
      </span>
    </span>
  </div>
</template>

<script>

import $_ from '../../utils/dashUtil.ts';

export default {
  name: 'GTooltip',
  props: {
    text: {
      type: String,
      default: '',
    },
    variant: {
      type: String,
      default: 'dark',
    },
    placement: {
      type: String,
      default: 'right',
    },
  },
  computed: {
    tooltipClass () {
      return {
        [$_(this.variant)]: true,
      };
    },
    placementClass () {
      return {
        [$_(this.placement)]: true,
        [$_(this.variant)]: true,
      };
    },
    borderColorVars () {
      if (this.variant) {
        switch (this.variant) {
          case 'primary':
            return { '--border-color': 'var(--orange-500)' };
          case 'secondary':
            return { '--border-color': 'var(--main-grey-500)' };
          case 'tertiary':
            return { '--border-color': 'var(--light-grey-500)' };
          case 'success':
            return { '--border-color': 'var(--green-500)' };
          case 'info':
            return { '--border-color': 'var(--blue-500)' };
          case 'danger':
            return { '--border-color': 'var(--red-500)' };
          case 'dark':
            return { '--border-color': 'var(--main-grey-500)' };
          default:
            return { '--border-color': 'var(--main-grey-500)' };
        }
      }
      return this.variant && { '--border-color': 'var(--orange-500)' };
    },
  },
};
</script>

<style lang="scss" scoped>
.g-tooltip {
  position: relative;
  display: inline-block;

  .text {
    position: absolute;
    font-family: var(--font-family-base);
    z-index: var(--zindex-tooltip);
    top: 50%;
    transform: translateY(-50%);
    left: 100%;
    max-width: 200px;
    max-height: 200px;
    text-align: left;
    visibility: hidden;
    opacity: 0;
    transition: opacity 0.4s ease;

    & > i.arrow {
      position: absolute;
      top: 50%;
      left: -2px;
      display: inline-block;
      transform: translateY(-50%);

      &.light {
        &:before {
          border: solid #fff;
        }
      }

      &:before {
        position: relative;
        content: '';
        border-style: solid;
        display: inline-block;
        padding: 3px;
      }
    }

    & > span {
      display: inline-block;
      color: #f1f1f1;
      border-radius: 4px;
      padding: 8px 12px;
      margin-top: 5px;
    }

    &.-success {
      i:before {
          border-color: var(--green-500);
      }
      & > span {
        background: var(--green-500);
      }
    }

    &.-primary {
      i:before {
          border-color: var(--orange-500);
      }
      & > span {
        background: var(--orange-500);
      }
    }

    &.-secondary {
      i:before {
          border-color: var(--main-grey-500);
      }
      & > span {
        background: var(--main-grey-500);
      }
    }

    &.-tertiary {
      i:before {
          border-color: var(--light-grey-500);
      }
      & > span {
        background: var(--light-grey-500);
      }
    }

    &.-info {
      i:before {
          border-color: var(--blue-500);
      }
      & > span {
        background: var(--blue-500);
      }
    }

    &.-warning {
      i:before {
          border-color: var(--orange-500);
      }
      & > span {
        background: var(--orange-500);
      }
    }

    &.-danger {
      i:before {
          border-color: var(--red-500);
      }
      & > span {
        background: var(--red-500);
      }
    }

    &.-light {
      i:before {
          border-color: var(--bg-grey-500);
      }
      & > span {
        background: var(--bg-grey-500);
        color: var(--main-grey-500);
      }
    }

    &.-dark {
      i:before {
          border-color: var(--main-grey-500);
      }
      & > span {
        background: var(--main-grey-500);
      }
    }

    &.-right {
      margin-left: 10px;

      & > i.arrow:before {
        border-width: 3px 0 0 3px;
        transform: rotate(-45deg);
      }

      &.start {
        top: 0;
        transform: translateY(0);
        & > i.arrow {
          top: 0;
          transform: translateY(0);
        }
      }

      &.end {
        top: initial;
        bottom: 0;
        transform: translateY(0);
        & > i.arrow {
          top: initial;
          bottom: 0;
          transform: translateY(0);
        }
      }
    }

    &.-left {
      margin-right: 10px;
      left: initial;
      right: 100%;

      & > i.arrow {
        left: initial;
        right: -2px;

        &:before {
          border-width: 3px 3px 0 0;
          transform: rotate(45deg);
        }
      }

      &.start {
        top: 0;
        transform: translateY(0);
        & > i.arrow {
          top: 0;
          transform: translateY(0);
        }
      }

      &.end {
        top: initial;
        bottom: 0;
        transform: translateY(0);
        & > i.arrow {
          top: initial;
          bottom: 0;
          transform: translateY(0);
        }
      }
    }

    &.-top {
      margin-bottom: 10px;
      top: initial;
      bottom: 100%;
      left: 50%;
      width: auto;
      transform: translate(-50%, 0);

      & > i.arrow {
        top: initial;
        left: 50%;
        bottom: -5px;
        transform: translate(-50%, 0);
        &:before {
          border-width: 0 3px 3px 0;
          transform: rotate(45deg);
        }
      }

      &.start {
        & > i.arrow {
          left: 5px;
          transform: translateX(0);
        }
      }

      &.end {
        & > i.arrow {
          left: initial;
          right: 5px;
          transform: translateX(0);
        }
      }
    }

    &.-bottom {
      margin-top: 6px;
      top: 100%;
      left: 50%;
      width: auto;
      transform: translate(-50%, 0);

      & > i.arrow {
        top: -1px;
        left: 50%;
        transform: translate(-50%, 0);
        &:before {
          border-width: 3px 3px 0 0;
          transform: rotate(-45deg);
        }
      }

      &.start {
        left: 0;
        transform: translate(0, 0);

        & > i.arrow {
          left: 5px;
          transform: translateX(0);
        }
      }

      &.end {
        right: 0;
        transform: translate(0, 0);
        & > i.arrow {
          left: initial;
          right: 5px;
          transform: translateX(0);
        }
      }
    }
  }

  &:hover {
    .text {
      visibility: visible;
      opacity: 1;
    }
  }
}
</style>
