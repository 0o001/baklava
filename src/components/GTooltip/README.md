# GTooltip

## Slots

| Name              | Description |
| ----------------- | ----------- |
| `tooltip-trigger` |             |
| `tooltip-text`    | &nbsp;      |

## Props

| Name        | Type     | Description | Default   |
| ----------- | -------- | ----------- | --------- |
| `text`      | `String` |             | `""`      |
| `variant`   | `String` |             | `"dark"`  |
| `placement` | `String` |             | `"right"` |

## Computed Properties

| Name              | Type      | Description                              |
| ----------------- | --------- | ---------------------------------------- |
| `tooltipClass`    | `object`  | **Dependencies:** `variant`              |
| `placementClass`  | `object`  | **Dependencies:** `placement`, `variant` |
| `borderColorVars` | `boolean` | **Dependencies:** `variant`              |

