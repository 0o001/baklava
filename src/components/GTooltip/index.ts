import GTooltip from './GTooltip.vue';

export default GTooltip;