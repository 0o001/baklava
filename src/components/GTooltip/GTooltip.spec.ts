import { shallowMount } from '@vue/test-utils';
import GTooltip from './';

describe('GTooltip', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(GTooltip, {
        propsData: {
            text: 'Tooltip Text',
        },
    });
  });

  it('should match snapshot when show is prop is given as false ', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render base variant as default when variant prop does not set', () => {
    expect(wrapper.vm.borderColorVars).toEqual({ '--border-color': 'var(--main-grey-500)' });
  });

  it('should render given variant when variant prop set', async () => {
    // given
    wrapper.setProps({
      variant: 'success',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.borderColorVars).toEqual({ '--border-color': 'var(--green-500)' });
  });

    it('shows the text if the tooltip trigger text is over or not', () => {
        wrapper.find('.g-tooltip').trigger('mouseover');
        wrapper.vm.$nextTick( () => {
            expect(wrapper.find('.text').text()).toContain('Tooltip Text');
        });
    });
});
