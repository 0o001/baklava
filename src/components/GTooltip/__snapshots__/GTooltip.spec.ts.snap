// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GTooltip should match snapshot when show is prop is given as false  1`] = `
<div
  class="g-tooltip"
>
   
  <span
    class="text -right -dark"
  >
    <i
      class="arrow"
    />
     
    <span>
      
      Tooltip Text
      
    </span>
  </span>
</div>
`;
